Source/Core/AST/CallCmd.cs

```diff
@@ -395,6 +395,20 @@ void CheckModifies(IEnumerable<Variable> modifiedVars)
           }
         }
       }
+      var a = calleeDecl.RefinedAction;
+      if (!IsAsync) {
+          while(a != null){
+            if (a.ActionDecl.RefinedAction != null) {
```

Comment:
Factor out the expression ```a.ActionDecl.RefinedAction``` into a separate local variable and use it in subsequent code.